servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpClientConfig.java

```diff
@@ -119,6 +137,20 @@ public int peerPort() {
                 public String sniHostname() {
                     return sniHostname;
                 }
+
+                @Nullable
+                private String evaluatePeerHost() {
+                    return (configPeerHost == null && inferPeerHost) ? fallbackPeerHost : configPeerHost;
```

Comment:
consider inlining these definitions or making the anonymous type non-anonymous and adding constructor to clarify we don't need to retain references to `configPeerHost` and related variables outside the scope of this class.